### PR TITLE
Add unit tests for HandleDuplicateHold

### PR DESCRIPTION
**Pull request by MarcinSwierczynski**
This a PR made as a part of Szkola Testow challenge.

The project is so good it was not easy to find a gap :) Hopefully, this is useful.

### CODE REVIEW

File: src/test/groovy/io/pillopl/library/lending/patron/application/hold/HandleDuplicateHoldTest.groovy [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import spock.lang.Specification
+
+import static io.pillopl.library.lending.book.model.BookFixture.anyBookId
+import static io.pillopl.library.lending.book.model.BookFixture.bookOnHold
+import static io.pillopl.library.lending.librarybranch.model.LibraryBranchFixture.anyBranch
+import static io.pillopl.library.lending.patron.model.PatronFixture.anyPatronId
+import static io.pillopl.library.lending.patron.model.PatronFixture.regularPatron
+import static java.time.Instant.now
+
+class HandleDuplicateHoldTest extends Specification {
```
Comment (zuzannapacholczyk) [inline src/test/groovy/io/pillopl/library/lending/patron/application/hold/HandleDuplicateHoldTest.groovy]: Looks good :)

**Comment by MarcinSwierczynski** [comment]
@pilloPl advised me to change the test to more like a mock-based approach because we're testing the business process here, not a domain logic. That's why I also had to externalize the `Clock`.

File: src/test/groovy/io/pillopl/library/lending/patron/application/hold/HandleDuplicateHoldTest.groovy (hunk 1 above)
Comment (MarcinSwierczynski) [reply src/test/groovy/io/pillopl/library/lending/patron/application/hold/HandleDuplicateHoldTest.groovy]: Thank you for taking a look!

**Comment by pilloPl** [comment]
Fantastic job. thanks @MarcinSwierczynski, @zuzannapacholczyk

**Pull request merged**